homeassistant/components/todo/__init__.py

```diff
@@ -77,6 +77,19 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
         _async_remove_todo_items,
         required_features=[TodoListEntityFeature.DELETE_TODO_ITEM],
     )
+    component.async_register_entity_service(
+        "list_items",
+        cv.make_entity_service_schema(
+            {
+                vol.Optional("status", default=[TodoItemStatus.NEEDS_ACTION]): vol.All(
+                    cv.ensure_list,
+                    [vol.In({TodoItemStatus.NEEDS_ACTION, TodoItemStatus.COMPLETED})],
+                ),
+            }
```

Comment:
I think we need to remove the default here from the service schema, and only add the default to the service description.

That way, the default will be set when using the UI (filter by needs action), but providing no status filter will return the complete list.

Calling the service with any filter, should not just go and filter by its own. Additionally, if you want all, you'd now have to provide a status filter to get to that (which is odd).

I get the intention of the default, but we should set that as a UI feature instead.

Reply:
Great catch, thank you for the thorough explanation. I have moved the default to `services.yaml`.